src/levels/level_1.gd

```diff
@@ -20,3 +22,9 @@ func puzzle_completed() -> void:
 	$Player.unfreeze()
 	$PuzzleConsole.hide()
 	$Door.open()
+
+
+func level_trigger_entered(_body: Node2D) -> void:
+	$LevelTransition.close()
+	await $LevelTransition.animation_finished
+	get_tree().change_scene_to_file("res://src/menu/root_menu.tscn")
```

Comment:
`level_1.gd` is starting to get more complicated and I'm wondering if we should be making stuff in here in the base level class or make components more reusable.

This right here should definitely be in its own reusable code block and not require any kind of signal connections or anything. Make a new `level_transition.gd` script which extends Area2D, has an @export annotation on what scene to load, and it will perform a level transition and load the next scene on finishing the transition